### DISCUSSION TITLE
Subscriptions via WebSockets close when opened

### DISCUSSION THREAD

**Discussion by lbkulinski** [Netflix/dgs-framework#1845] [Q&A]
Hello!

I am trying to add a subscription via WebSockets. When I try to connect via `ws://localhost:8080/subscriptions` in Postman, I see that the connection is established, but then quickly closes. Upon further inspection, Postman shows `1002 An unrecoverable IOException occurred so the connection was closed`.

I have the following dependencies for the DGS framework in my `pom.xml`:
```
<dependencyManagement>
    <dependencies>
         <dependency>
             <groupId>com.netflix.graphql.dgs</groupId>
             <artifactId>graphql-dgs-platform-dependencies</artifactId>
             <version>8.4.1</version>
             <type>pom</type>
             <scope>import</scope>
        </dependency>
    </dependencies>
</dependencyManagement>
...
<dependency>
    <groupId>com.netflix.graphql.dgs</groupId>
     <artifactId>graphql-dgs-spring-boot-starter</artifactId>
</dependency>
<dependency>
    <groupId>com.netflix.graphql.dgs</groupId>
    <artifactId>graphql-dgs-subscriptions-websockets-autoconfigure</artifactId>
</dependency>
```

My schema is as follows:
```
enum Line {
    RED
    BLUE
    BROWN
    GREEN
    ORANGE
    PURPLE
    PINK
    YELLOW
}

type Train {
    run: Int!
    line: Line!
    destination: String!
    station: String!
    predictionTime: String!
    arrivalTime: String!
    due: Boolean!
    scheduled: Boolean!
    delayed: Boolean!
}

type Subscription {
    trainArrivals(stationId: ID!): [Train!]!
}
```

My handler method is as follows:
```
@DgsSubscription
    public Publisher<List<Train>> trainArrivals(@InputArgument String stationId) {
        Objects.requireNonNull(stationId);

        Duration duration = Duration.ofMinutes(1L);

        return Flux.interval(duration)
                   .flatMap(tick -> Flux.just(this.client.getTrains(stationId)))
                   .map(TrainResponse::body)
                   .map(TrainBody::trains);
}
```

When making the request, I don't see any errors in the console. If seeing the full application would be helpful, the repository can be found here: https://github.com/lbkulinski/cta4j-back-end.

Thanks!

**Comment by srinivasankavitha** [comment]
Not sure about Postman, but you could alternatively try using graphiql for testing your subscription and set breakpoint to debug through the subscriptions data fetcher. The dependencies look fine to me. We also have an example of a simple subscription that uses websockets here: https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-example-shared/src/main/java/com/netflix/graphql/dgs/example/shared/datafetcher/SubscriptionDataFetcher.java#L28

Hope that helps.

**Reply by lbkulinski** [reply]
Do subscriptions work with graphiql? I did try using it to debug, but they seem to not. I also tried to hook this up to my existing front-end which uses the Apollo client, and also ran into issues. I believe I've seen this example, but I will take another look.

Could there be some sort of dependency versioning issue? I am using the latest versions of Spring Boot deps with Maven. Thanks!

**Reply by lbkulinski** [reply]
Oddly, after removing the following dependencies:
```
 <dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-core</artifactId>
    <version>2.17.0</version>
</dependency>
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.17.0</version>
</dependency>
<dependency>
    <groupId>com.fasterxml.jackson.datatype</groupId>
    <artifactId>jackson-datatype-jsr310</artifactId>
    <version>2.17.0</version>
</dependency>
```
the subscription worked. Any idea why this would be?

**Reply by srinivasankavitha** [reply]
Not sure, but very likely it was puling in conflicting versions of these deps. Subscriptions do depend on them so maybe it messed up deserialization/serialization of request/responses and affected the functionality.

Glad it's working for you in any case.
